File: blue_star.rb
class BlueStar
  attr_reader :award

  def initialize(award)
    @award = award
  end

  def update
    award.expires_in -= 1
    return if award.quality.zero?

    award.quality -= 2
    award.quality -= 2 if award.expires_in <= 0
  end
end

File: blue_star.py
from award import Award

class BlueStar(Award):
	def __init__(self, expires_in=None, quality=None):
		super(BlueStar, self).__init__('Blue Star', expires_in, quality)

	def update_quality(self):
		if self.quality > 0:
			self.quality -= 1
		if self.quality > 0:
			self.quality -= 1
		self.expires_in -= 1
		if self.expires_in < 0 and self.quality > 0:
			self.quality -= 1
		if self.expires_in < 0 and self.quality > 0:
			self.quality -= 1